### DISCUSSION TITLE
how to extract all values of same key from json array?

### DISCUSSION THREAD

**Discussion by xkungfu** [nlohmann/json#3213] [Q&A] [answered]
how to extract all values of same key from json array?

```
json j =
[
{"code":"B1","name":"jhon","date":"2021-12-16"},
{"code":"B2","name":"smith","date":"2021-12-17"},
{"code":"B3","name":"someone","date":"2021-12-18"},
{"code":"B4","name":"jhon","date":"2021-12-19"},
{"code":"B5","name":"mary","date":"2021-12-20"}
]
```

I want to a result like below:

`["jhon", "smith", "someone", "mary"]
`

I can get the result through a loop.

but is there any other simple way?   
such as: j.get_all_values["name"]

**Comment by nlohmann** [comment, marked as answer]
There is no dedicated function for this in the library.

**Reply by xkungfu** [reply]
I see!
now I have to loop it.
hope an api for this.
thank you!
